src/classes/UTIL_CurrencyCache.cls

```diff
@@ -77,6 +77,12 @@ public class UTIL_CurrencyCache implements UTIL_CurrencyCache.Interface_x {
             // If a set of rates was retrieved, find the effective rate based on the date
             if (cachedMapOfRatesByCurrencyCode.containsKey(currCode)) {
                 UTIL_CurrencyCache.CurrencyData currData = cachedMapOfRatesByCurrencyCode.get(currCode);
+                
+                //In this case the currency was likely disabled and the rates are null e.g. {AUD=null} 
+                //Return 1, to align with existing logic.
+                if(currData == null){
```

Comment:
Being super pedantic here but our coding standard has a space between the `)` and the `{`.  We should check out Uncrustify for our code formatting, or you might know of something even better.